### ISSUE TITLE
[Feature] OSK (Onscreen Keyboard) - Adjustable Size

### ISSUE THREAD

**Issue by mindinsomnia** [ValveSoftware/SteamOS#904]
The OSK by default right now takes up about 50% of the screen.

I feel as a regular user of the OSK, typing using the touchpads on the side of the Deck, I really don't need the OSK to be quite so large, I could quite easily see which keys are highlighted even if the OSK was as little as only 50% of the current size.

I think it would be great if we had a slider to adjust the size of the OSK.

**Comment by TOGLK** [comment]
I agree that at times the keyboard feels far too big for the size of the display.

## How to tackle the problem?
I feel like a suitable way to tackle this would be to have two variables that can be adjusted by the user in the settings for the On-Screen Keyboard.

### Variable No.1
The first Variable would be "Width", the minimum width you'd be able to scale the keyboard down to would be 30% of it's original size, but by doing so any width reduction below 60% would also start to scale down the height as well to ensure that all characters of the keyboard remain visible on-screen still.

### Variable No.2
The second variable would be "Height", the minimum height you would be able to set would be 30% of the default On-Screen Keyboard Height, but, unlike the "Width" variable the Height of the keyboard would scale without any tie-in to the width, this would allow users to take full advantage of the full width of the display whilst reducing the overall height and still having a full-sized default layout be visible & accessible.


## Additional Thoughts
It has come to my mind that perhaps with the addition of overall On-screen Keyboard "Width" & "Height" parameters being adjustable that it might also be worth providing end-users with the option to choose the width of particular keys on the keyboard so that keys used most often can be more easily interacted with.

This could be useful for Function Row Keys (F1-F24) as well as W, A, S, &, D Keys.
Maybe this is something to also be considered by the team at Valve.
